packages/extension-browser/src/content-script/fetcher.ts

```diff
@@ -108,6 +110,8 @@ export class Fetcher {
     private resolveFetch(event: FetchEnd) {
```

Comment:
To make this work in the Firefox code-path, we should replace the following above (around line 70) with a call to this `resolveFetch` method instead.

```diff
-                if (this._fetches.has(target) && response.type === 'basic') {
+                if (response.type === 'basic') {
                    // Firefox code-path (can't use other path as fetch doesn't trigger onRequestFinished).
                    const data = await toNetworkData(target, headers, response);

-                    resolve(data);
-                    this._fetches.delete(target);
+                    this.resolveFetch(data);
                } else {
```

Reply:
Resolving as we discussed this data is already set by the `toNetworkData` helper.